modules/programs/obsidian.nix

```diff
@@ -0,0 +1,547 @@
+    "workspaces"
+    "zk-prefixer"
+  ];
+in
+{
+  options.programs.obsidian = {
+    enable = mkEnableOption "obsidian";
+    package = mkPackageOption pkgs "obsidian" { };
+
+    defaultSettings = {
+      app = mkOption {
+        description = "Settings to write to app.json.";
+        type = types.raw;
+        default = { };
+      };
+
+      appearance = mkOption {
+        description = "Settings to write to appearance.json.";
+        type = types.raw;
+        default = { };
+      };
+
+      corePlugins = mkOption {
+        description = "Core plugins to activate.";
+        type = types.raw;
+        default = [
+          "backlink"
+          "bookmarks"
+          "canvas"
+          "command-palette"
+          "daily-notes"
+          "editor-status"
+          "file-explorer"
+          "file-recovery"
+          "global-search"
+          "graph"
+          "note-composer"
+          "outgoing-link"
+          "outline"
+          "page-preview"
+          "switcher"
+          "tag-pane"
+          "templates"
+          "word-count"
+        ];
+      };
+
+      communityPlugins = mkOption {
+        description = "Community plugins to install and activate.";
+        type = types.raw;
+        default = [ ];
+      };
+
+      cssSnippets = mkOption {
+        description = "CSS snippets to install.";
+        type = types.raw;
+        default = [ ];
+      };
+
+      themes = mkOption {
+        description = "Themes to install.";
+        type = types.raw;
+        default = [ ];
+      };
+
+      hotkeys = mkOption {
+        description = "Hotkeys to configure.";
+        type = types.raw;
+        default = { };
+      };
+
+      extraFiles = mkOption {
+        description = "Extra files to link to the vault directory.";
+        type = types.raw;
+        default = { };
+      };
+    };
+
+    vaults = mkOption {
+      description = "List of vaults to create.";
+      type = types.attrsOf (
+        types.submodule (
+          { name, config, ... }:
+          {
+            options = {
+              enable = mkOption {
+                type = types.bool;
+                default = true;
+                description = "Whether this vault should be generated.";
+              };
+
+              target = mkOption {
+                type = types.str;
+                defaultText = literalExpression "name";
+                description = "Path to target vault relative to the user's {env}`HOME`.";
+              };
+
+              settings = {
+                app = mkOption {
+                  description = "Settings to write to app.json.";
+                  type = with types; attrsOf anything;
+                  default = cfg.defaultSettings.app;
+                };
+
+                appearance = mkOption {
+                  description = "Settings to write to appearance.json.";
+                  type = with types; attrsOf anything;
+                  default = cfg.defaultSettings.appearance;
```

Comment:
Oh I see... so a vault can have settings specific to them and it by default just inherits the top level default settings?